salt/transport/zeromq.py

```diff
@@ -905,12 +908,14 @@ def publish(self, load):
             'Sending payload to publish daemon. jid=%s size=%d',
             load.get('jid', None), len(payload),
         )
-        pub_sock.send(payload)
+        if msg_queue:
+            msg_queue.put(payload)
+        else:
+            pub_sock.send(payload)
+            pub_sock.close()
```

Comment:
should we cache the socket and not create it and close it for each message we publish?

Reply:
I did try storing the context and socket on the class so it won't get created each time but I still saw the issue. @kingsquirrel152 also talks about trying that as a fix in #36469

Reply:
I fixed this issue in an older version of salt stack (back when it was single threaded and not driven by tornado). The issue iirc is that creating the socket, and sending data, then _tearing the socket down_, breaks the contract with zmq. You have no guarantee the message is ever sent. I think there are some settings for the socket that could help (linger for instance). TLDR though - I think your best bet is to follow @rares-pop suggest and cache the socket avoiding the issue entirely.